drivers/metal/rendering_shader_container_metal.mm

```diff
@@ -0,0 +1,706 @@
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.storage_buffers.empty()) {
+			Error err = descriptor_bindings(resources.storage_buffers, Writable::Maybe);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.storage_images.empty()) {
+			Error err = descriptor_bindings(resources.storage_images, Writable::Maybe);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.sampled_images.empty()) {
+			Error err = descriptor_bindings(resources.sampled_images, Writable::No);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.separate_images.empty()) {
+			Error err = descriptor_bindings(resources.separate_images, Writable::No);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.separate_samplers.empty()) {
+			Error err = descriptor_bindings(resources.separate_samplers, Writable::No);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+		if (!resources.subpass_inputs.empty()) {
+			Error err = descriptor_bindings(resources.subpass_inputs, Writable::No);
+			ERR_FAIL_COND_V(err != OK, false);
+		}
+
+		if (!resources.push_constant_buffers.empty()) {
+			for (Resource const &res : resources.push_constant_buffers) {
+				uint32_t binding = compiler.get_automatic_msl_resource_binding(res.id);
+				if (binding != (uint32_t)-1) {
+					stage_data.push_constant_binding = binding;
+				}
+			}
+		}
+
+		ERR_FAIL_COND_V_MSG(!resources.atomic_counters.empty(), false, "Atomic counters not supported");
+		ERR_FAIL_COND_V_MSG(!resources.acceleration_structures.empty(), false, "Acceleration structures not supported");
+		ERR_FAIL_COND_V_MSG(!resources.shader_record_buffers.empty(), false, "Shader record buffers not supported");
+
+		if (!resources.stage_inputs.empty()) {
+			for (Resource const &res : resources.stage_inputs) {
+				uint32_t binding = compiler.get_automatic_msl_resource_binding(res.id);
+				if (binding != (uint32_t)-1) {
+					stage_data.vertex_input_binding_mask |= 1 << binding;
+				}
+			}
+		}
+
+		stage_data.is_position_invariant = compiler.is_position_invariant();
+		stage_data.supports_fast_math = !entry_point.flags.get(spv::ExecutionModeSignedZeroInfNanPreserve);
+		stage_data.hash = SHA256Digest(source.c_str(), source.length());
+		stage_data.source_size = source.length();
+		::Vector<uint8_t> binary_data;
+		binary_data.resize(stage_data.source_size);
+		memcpy(binary_data.ptrw(), source.c_str(), stage_data.source_size);
+
+		if (export_mode) {
+			// Try to compile the Metal source code
+			::Vector<uint8_t> library_data;
+			Error compile_err = compile_metal_source(source.c_str(), stage_data, library_data);
+			if (compile_err == OK) {
+				stage_data.library_size = binary_data.size();
+				binary_data.resize(stage_data.source_size + stage_data.library_size);
+				memcpy(binary_data.ptrw() + stage_data.source_size, library_data.ptr(), stage_data.library_size);
+			}
+		}
+
+		uint32_t binary_data_size = binary_data.size();
```

Comment:
binary_data.size() should be library_data.size()

The current line records the wrong size for the prebuilt metallib.

Reply:
library_data is not in the scope of this line. Are you sure you don't mean to correct this one instead?

```stage_data.library_size = binary_data.size();```

Reply:
Sorry, I gave the comment for the wrong line.